torch/distributed/distributed_c10d.py

```diff
@@ -1614,7 +1614,7 @@ def all_gather_multigpu(
 
 def _object_to_tensor(obj, device):
     f = io.BytesIO()
-    _pickler(f).dump(obj)
+    torch.save(obj,f,pickle_protocol=4)
```

Comment:
IIUC, to have semantic parity, we want:
```suggestion
    torch.save(obj, f, pickle_protocol=pickle.DEFAULT_PROTOCOL)
```
`_pickler = pickle.Pickler`, and we call `_pickler(f)` without a protocol specified, which means it uses the default protocol.

`DEFAULT_PROTOCOL` is only raised to `4` in 3.8: https://docs.python.org/3/library/pickle.html#pickle.DEFAULT_PROTOCOL